lib/meilisearch-rails.rb

```diff
@@ -794,7 +796,9 @@ def meilisearch_settings_changed?(prev, current)
           prev_v = prev[k.to_s]
           if v.is_a?(Array) && prev_v.is_a?(Array)
             # compare array of strings, avoiding symbols VS strings comparison
-            return true if v.map(&:to_s) != prev_v.map(&:to_s)
+            return true if (v.map(&:to_s) - prev_v.map(&:to_s)).length.positive?
```

Comment:
@yagince the order matters for some of the configurations like `relevancy`. Even if the array contains the same data like `['words', 'typo']` in the engine, they have a different behavior than `['typo', 'words']`
https://docs.meilisearch.com/learn/core_concepts/relevancy.html#behavior

Reply:
@brunoocasali 

Indeed. I did not take it into consideration enough.
Thank you.

I think the only setting where the order matters is `rankingRules`, is that right?
If so, I would exclude only `rankingRules`.

Reply:
Why don't you go back to the old behavior? I think the code will be simpler to understand and less error-prone, especially if we adopt new settings with that behavior (the code will don't have to rely on someone else memory to keep it working).

PS: I understand the issue regarding the performance side, but does the array order matter that much? Should meilisearch-rails care about that, or should the app handle this?

Reply:
Settings such as `searchableAttributes` will change the order of values that can be retrieved from meilisearch, even if the settings on the meilisearch-rails side have not changed.
This causes meilisearch-rails to assume that the settings have changed and send a request to meilisearch to update the settings.
When this update is performed, meilisearch indexing becomes very slow and indexing will not be performed for a while.
This is the problem I am having.

I do not want to make meilisearch-rails complicated.
(I don't think it is a good idea to change the behavior for each key in settings.)
I was thinking that I should fix meilisearch itself, but I found a place where meilisearch-rails checks for changes in settings.
So I understood that meilisearch-rails's policy is to check changes on the library side.
I thought it would be better to check for changes in the settings on the meilisearch-rails side, but it is more complicated than I thought.
I think I should close this PR and I should think another idea.